### PR TITLE
Drop support for active record < 4.2, resolves #4

### LINKED ISSUE #4: Ruby 2.4 incompatible with AR 4.0 & 4.1
So yes, there is no way to use AR below 4.2 with Ruby 2.4+. AR fixes the Integer (Fixnum + Bignum) in Arel 6 which is used in AR 4.2 https://github.com/rails/arel/commit/1104b6671251dfa72427210e6336c0c01dc23b5f

You could loose support for AR < 4.2 or perhaps add a note explaining that if using AR 4.1 you cannot use Ruby above `~> 2.3`. This is an active record issue anyway, so no one will be using these combinations. You could just update the test matrix to simulate what I have just described.

### PR DESCRIPTION
**Pull request by mcelicalderon**
The readme update is just an idea, not sure if needed as Active Record itself won't work with the ruby and AR versions removed from the build.

### CODE REVIEW

**Comment by mcelicalderon** [comment]
I added a new appraisal for AR 4.0.6 which I tested is the lowest AR that doesn't break the build. Also added `spec.add_dependency 'activerecord', ['>= 4.0.6', '< 5.3']` as the build would break with anything lower than 4.6.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,14 @@
 language: ruby
 
 rvm:
-  - 2.3.0
-  - 2.3.3
-  - 2.4.0
+  - 2.2.10
```
Comment (robinroestenburg) [inline .travis.yml]: I don't want to support Ruby 2.2 as it is no longer supported: https://www.ruby-lang.org/en/news/2018/06/20/support-of-ruby-2-2-has-ended/

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+  - 2.3.7
+  - 2.4.4
+  - 2.5.1
   - ruby-head
 
 gemfile:
+  - gemfiles/rails_40.6.gemfile
```
Comment (robinroestenburg) [inline .travis.yml]: I appreciate the effort of checking what the minimal version of Rails is that can be run in combination with `deffering`, but let's just remove support for Rails 4.0 and Rails 4.1 all together.

Two reasons:
- http://guides.rubyonrails.org/maintenance_policy.html#severe-security-issues  
Patches for severe security issues are no longer provided by Rails for these versions.
- http://blog.scoutapp.com/articles/2017/01/09/state-of-the-2017-rails-stack  
These versions are not being used a lot (blog was from early 2017, so probably less now).

Could you reduce the appraisals to:
- rails-42
- rails-50
- rails-51
- rails-52

File: deferring.gemspec [hunk 3]
```diff
@@ -21,7 +20,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'activerecord', ['> 4.0', '< 5.3']
+  spec.add_dependency 'activerecord', ['>= 4.0.6', '< 5.3']
```
Comment (robinroestenburg) [inline deferring.gemspec]: Update the gemspec to reflect the minimum Rails version of 4.2.

**Comment by mcelicalderon** [comment]
No problem, support dropped ;)

**Comment by mcelicalderon** [comment]
No problem @robinroestenburg happy to help. I've been using this for a couple months and can't understand how this is not the default rails behavior, it should.

**Comment by robinroestenburg** [comment]
@mcelicalderon I've just pushed version 0.5.0 of the gem which adds the support for 5.2 (and drops the 4.0 and 4.1 support).

> I've been using this for a couple months and can't understand how this is not the default AR behavior, it should.

Yea, I am amazed by that too. Years ago we did try to get it merged, but it was rejected :-/ We've been using it in production for about 4 years now, it makes everything a lot simpler.

Glad you like it. Just let me know if you have any feature requests or run into any bugs.

**Pull request merged**